core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/guess.kt

```diff
@@ -35,7 +37,7 @@ public sealed interface SupportedFormat {
 public sealed interface SupportedFormatSample {
 
     @JvmInline
-    public value class DataFile(public val sampleFile: File) : SupportedFormatSample
```

Comment:
I hope this doesn't cause binary compatibility issues between beta3 and beta4, but I fear it does. Maybe you can better add `DataPath`?